service/src/main/java/de/adorsys/psd2/sandbox/portal/testdata/TestDataService.java

```diff
@@ -1,209 +1,294 @@
 import java.math.BigDecimal;
 import java.time.LocalDate;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.Currency;
 import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 import org.springframework.stereotype.Service;
 
 @Service
 public class TestDataService {
 
-  public static final String PSU_ID = "PSU-1";
-  public static final String PSU_PASSWORD = "12345";
-  public static final String IBAN_GIRO = "DE94500105178833114935";
-  public static final String IBAN_SAVINGS = "DE96500105179669622432";
-  public static final Currency EUR = Currency.getInstance("EUR");
-  public static final String ACCOUNT_OWNER = "Melanie Klein";
-  public static final String ACCOUNT_ID_GIRO = "9b86539d-589b-4082-90c2-d725c019777f";
-  public static final String ACCOUNT_ID_SAVINGS = "d460057b-053a-490a-a36e-c0c8afb735e9";
+  private static final Currency EUR = Currency.getInstance("EUR");
 
-  private HashMap<String, Account> accounts;
-  private HashMap<String, HashMap<String, Transaction>> transactions;
+  private Map<String, PsuData> psuMap;
 
+  /**
+   * Default Constructor.
+   */
   public TestDataService() {
-    this.accounts = createAccounts();
-    this.transactions = createTransactions();
+    HashMap<String, PsuData> map = new HashMap<>();
+
+    PsuData psuSuccessful = initPsuSuccessfull();
+    map.put(psuSuccessful.getPsuId(), psuSuccessful);
+
+    this.psuMap = Collections.unmodifiableMap(map);
   }
 
-  public List<Transaction> getTransactions(String accountId) {
-    return new ArrayList<>(transactions.get(accountId).values());
+  public Optional<PsuData> getPsu(String psuId) {
+    return Optional.ofNullable(psuMap.get(psuId));
   }
 
-  public Transaction getTransactionDetails(String accountId, String transactionId) {
-    return transactions.get(accountId).get(transactionId);
+  /**
+   * Returns Psu-Id by Iban.
+   *
+   * @param iban Iban
+   * @return String Psu-Id
+   */
+  public Optional<String> getPsuByIban(String iban) {
+    for (Map.Entry<String, PsuData> currentPsu : psuMap.entrySet()) {
+      String currentPsuId = currentPsu.getValue().getPsuId();
+
+      List<Account> result = currentPsu.getValue().getAccounts().values().stream()
+          .filter(account -> account.getIban().equals(iban))
+          .collect(Collectors.toList());
+
+      if (result.size() > 0) {
+        return Optional.of(currentPsuId);
+      }
+    }
+    return Optional.empty();
   }
 
-  public List<Account> getAccounts() {
-    return new ArrayList<>(accounts.values());
+  /**
+   * Returns AccountId by Iban.
+   *
+   * @param psuId psuId
+   * @param iban  Iban
+   * @return AccountId
+   */
+  public Optional<String> getAccountIdByIban(String psuId, String iban) {
+    HashMap<String, Account> map = psuMap.get(psuId).getAccounts();
+
+    Account result = map.values().stream()
```

Comment:
This crashes if no account has the given IBAN

Reply:
Wrote new Unit-Tests and fixed it.